File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int maxLength = 0;
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < nums.length; i++ ){
            set.add(nums[i]);
        }

        for (int i = 0; i< nums.length; i++){
            if (set.contains(nums[i]- 1)){
                
                continue;
            }
            else{
                int length = 0;
                while (set.contains(nums[i] + length)){
                    length += 1;
                } 
                maxLength = Math.max(maxLength, length);
            }
        }
        return maxLength;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        const res = []
        const curWindow = []

        for (let i = 0; i < nums.length; i++){
            curWindow.push(nums[i])
            if(curWindow.length == k){
                res.push(Math.max(...curWindow))
                curWindow.shift()
            }
        }
        return res
    }
}
